help_to_heat/frontdoor/interface.py

```diff
@@ -613,9 +616,40 @@ def get_address_and_epc_lmk(self, building_name_or_number, postcode):
 
     def get_epc(self, lmk):
         epc_api = EPCApi()
-        epc_details = epc_api.get_epc_details(lmk)
-        epc_recommendations = epc_api.get_epc_recommendations(lmk)
-        return epc_details, epc_recommendations
+        epc_details_response = self._get_epc_details(epc_api, lmk)
+        epc_recommendations = self._get_epc_recommendations(epc_api, lmk)
+        return (
+            epc_details_response,
+            epc_recommendations,
+        )
+
+    def _get_epc_details(self, epc_api, lmk):
+        return epc_api.get_epc_details(lmk)["rows"][0]
+
+    RECOMMENDATION_RETRIES_COUNT = 3
+
+    def _get_epc_recommendations(self, epc_api, lmk):
+        for i in range(self.RECOMMENDATION_RETRIES_COUNT):
```

Comment:
Not really necessary to change, but I'd instinctively have expected this sort of thing to be a while loop with an incrementing counter. Suppose it means we don't have to create this range (technically more scalable if we were doing colossal numbers of retries).

Reply:
for what it's worth range in python 3 [is lazy](https://stackoverflow.com/questions/23725258/does-range-involve-eager-or-lazy-evaluation-in-2-x-and-3-x), so it'll probably be using a counter behind the scenes. I don't think it'll ever construct the range unless we call `list()` or similar?

Reply:
Yeah I was wondering whether it ends up being equivalent once you get down to internals.